main/src/main/java/org/mobilitydata/gtfsvalidator/validator/DuplicateFareRuleZoneIdFieldsValidator.java

```diff
@@ -77,9 +78,18 @@ public void validate(NoticeContainer noticeContainer) {
    * @return the hashcode associated to the combination of this {@code GtfsFareRule} "route_id",
    *     "origin_id", "contains_id" and "destination_id".
    */
-  private int getHash(GtfsFareRule fareRule) {
-    return Objects.hash(
-        fareRule.routeId(), fareRule.originId(), fareRule.containsId(), fareRule.destinationId());
+  private static long getHash(GtfsFareRule fareRule) {
+    return HASH_FUNCTION
+        .newHasher()
+        .putUnencodedChars(fareRule.routeId())
```

Comment:
Why not using `putString` here?

Reply:
Yep, exactly! We care for speed, so we don't bother with putString that will reencode the string.